From da7239d27dd36ed264fad47e1051f0f82c9c17b0 Mon Sep 17 00:00:00 2001
From: cpw <cpw@weeksfamily.ca>
Date: Fri, 6 Nov 2015 15:16:23 -0500
Subject: [PATCH 114/122] Properly implement options for the renamer stuff.
 "JVN" argument enables JAD style variable naming (int i, j, k etc). Can pass
 a "RENAMER_FACTORY" to the decompiler context: this is the name of a class
 implementing IVariableNamingFactory, that will be loaded and instantiated,
 overriding other options.

---
 src/org/jetbrains/java/decompiler/main/ClassWriter.java   | 1 -
 .../jetbrains/java/decompiler/main/DecompilerContext.java | 8 ++++++--
 .../decompiler/main/extern/IAbstractParameterRenamer.java | 7 -------
 .../decompiler/main/extern/IFernflowerPreferences.java    | 3 +++
 4 files changed, 9 insertions(+), 10 deletions(-)
 delete mode 100644 src/org/jetbrains/java/decompiler/main/extern/IAbstractParameterRenamer.java

diff --git a/src/org/jetbrains/java/decompiler/main/ClassWriter.java b/src/org/jetbrains/java/decompiler/main/ClassWriter.java
index c515d2e..1338a42 100644
--- a/src/org/jetbrains/java/decompiler/main/ClassWriter.java
+++ b/src/org/jetbrains/java/decompiler/main/ClassWriter.java
@@ -18,7 +18,6 @@ package org.jetbrains.java.decompiler.main;
 import org.jetbrains.java.decompiler.code.CodeConstants;
 import org.jetbrains.java.decompiler.main.ClassesProcessor.ClassNode;
 import org.jetbrains.java.decompiler.main.collectors.BytecodeMappingTracer;
-import org.jetbrains.java.decompiler.main.extern.IAbstractParameterRenamer;
 import org.jetbrains.java.decompiler.main.extern.IFernflowerLogger;
 import org.jetbrains.java.decompiler.main.extern.IFernflowerPreferences;
 import org.jetbrains.java.decompiler.main.rels.ClassWrapper;
diff --git a/src/org/jetbrains/java/decompiler/main/DecompilerContext.java b/src/org/jetbrains/java/decompiler/main/DecompilerContext.java
index 8d4d403..d66145a 100644
--- a/src/org/jetbrains/java/decompiler/main/DecompilerContext.java
+++ b/src/org/jetbrains/java/decompiler/main/DecompilerContext.java
@@ -68,11 +68,15 @@ public class DecompilerContext {
       try {
         currentContext.get().renamerFactory = Class.forName((String) DecompilerContext.getProperty(RENAMER_FACTORY)).asSubclass(IVariableNamingFactory.class).newInstance();
       } catch (Exception e) {
-
+        getLogger().writeMessage("Error loading renamer factory class", e);
       }
     }
     if (DecompilerContext.getNamingFactory() == null) {
-      currentContext.get().renamerFactory = new JADNameProvider.JADNameProviderFactory();
+      if (DecompilerContext.getOption(IFernflowerPreferences.USE_JAD_VARNAMING)) {
+        currentContext.get().renamerFactory = new JADNameProvider.JADNameProviderFactory();
+      } else {
+        currentContext.get().renamerFactory = new IdentityRenamerFactory();
+      }
     }
   }
 
diff --git a/src/org/jetbrains/java/decompiler/main/extern/IAbstractParameterRenamer.java b/src/org/jetbrains/java/decompiler/main/extern/IAbstractParameterRenamer.java
deleted file mode 100644
index 4c509eb..0000000
--- a/src/org/jetbrains/java/decompiler/main/extern/IAbstractParameterRenamer.java
+++ /dev/null
@@ -1,7 +0,0 @@
-package org.jetbrains.java.decompiler.main.extern;
-
-import org.jetbrains.java.decompiler.main.rels.MethodWrapper;
-
-public interface IAbstractParameterRenamer {
-  public String renameParameter(String orig, int index, MethodWrapper wrapper, int flags);
-}
diff --git a/src/org/jetbrains/java/decompiler/main/extern/IFernflowerPreferences.java b/src/org/jetbrains/java/decompiler/main/extern/IFernflowerPreferences.java
index 79dd8e8..7940dcf 100644
--- a/src/org/jetbrains/java/decompiler/main/extern/IFernflowerPreferences.java
+++ b/src/org/jetbrains/java/decompiler/main/extern/IFernflowerPreferences.java
@@ -62,6 +62,8 @@ public interface IFernflowerPreferences {
 
   String INCLUDE_ENTIRE_CLASSPATH = "iec";
 
+  String USE_JAD_VARNAMING = "jvn";
+
   Map<String, Object> DEFAULTS = Collections.unmodifiableMap(new HashMap<String, Object>() {{
     put(REMOVE_BRIDGE, "1");
     put(REMOVE_SYNTHETIC, "0");
@@ -96,5 +98,6 @@ public interface IFernflowerPreferences {
     put(UNIT_TEST_MODE, "0");
     put(DUMP_ORIGINAL_LINES, "0");
     put(INCLUDE_ENTIRE_CLASSPATH, "0");
+    put(USE_JAD_VARNAMING, "0");
   }});
 }
-- 
2.21.0.windows.1

